renku/ui/service/views/v1/cache.py

```diff
@@ -84,9 +88,33 @@ def migration_check_project_view_1_5(user_data, cache):
       tags:
         - cache
     """
+
+    from flask import jsonify
+
+    from renku.ui.service.serializers.rpc import JsonRPCResponse
+    from renku.ui.service.views.error_handlers import pretty_print_error
+
     ctrl = MigrationsCheckCtrl(cache, user_data, dict(request.args), GitlabAPIProvider())
-    ctrl.RESPONSE_SERIALIZER = ProjectMigrationCheckResponseRPC_1_5()  # type: ignore
-    return ctrl.to_response()
+
+    if "project_id" in ctrl.context:  # type: ignore
+        result = asdict(ctrl.execute_op())
+    else:
+        # NOTE: use quick flow but fallback to regular flow in case of unexpected exceptions
+        try:
+            result = asdict(ctrl._fast_op_without_cache())
+        except (AuthenticationError, ProjectNotFound):
+            raise
+        except BaseException:
+            result = asdict(ctrl.execute_op())
+
+    if result.get("errors", None):
+        error = result["errors"][list(result["errors"].keys())[0]]
+        return jsonify(JsonRPCResponse().dump({"error": pretty_print_error(error)}))
+
+    else:
+        del result["errors"]
```

Comment:
This raises an error if `errors` isn't present. Is it always available? Maybe `result.pop('errors', None)`.